controllers/secret.go

```diff
@@ -24,187 +24,108 @@ import (
+	namespacedSecretName := newNamespacedName(secretName, meta.Namespace)
+	secret := &corev1.Secret{}
+	if !IsObjectFound(r.Client, namespacedSecretName, secret) {
 		err := errors.New(fmt.Sprintf("an existing secret named %s was not found in namespace %s", secretName, meta.Namespace))
 		log.Error(err, fmt.Sprintf("unable to reconcile %s in namespace %s", secretName, meta.Namespace))
 		return err
 	}
-	log.Info(fmt.Sprintf("existing %s secret found in namespace %s", secretName, meta.Namespace))
-	pyxisApiSecret := &corev1.Secret{}
-	if r.Client.Get(context.TODO(), key, pyxisApiSecret) != nil {
-		log.Error(err, fmt.Sprintf("unable to get the %s secret", secretName))
+	log.Info(fmt.Sprintf("an existing secret named %s was found in namespace %s", secretName, meta.Namespace))
+	secret = &corev1.Secret{}
+	err := r.Client.Get(context.TODO(), namespacedSecretName, secret)
+	if err != nil {
+		log.Error(err, fmt.Sprintf("unable to get the %s secret in the %s namespace", secretName, meta.Namespace))
 		return err
 	}
-	if pyxisApiSecret.Data[PYXIS_API_KEY] == nil || len(pyxisApiSecret.Data[PYXIS_API_KEY]) == 0 {
-		err = errors.New(fmt.Sprintf("the pyxis api key in %s is empty!", secretName))
-		log.Error(err, fmt.Sprintf("The %s secret does not contain the expected key", secretName))
+	log.Info(fmt.Sprintf("successfully fetched secret named %s in namespace %s", secretName, meta.Namespace))
+	err = errors.New(fmt.Sprintf("the %s secret does not contain a valid key %s", secretName, secretKeyName))
+	if value, ok := secret.Data[secretKeyName]; ok {
+		if value == nil || len(value) == 0 {
+			return err
+		}
+		log.Info(fmt.Sprintf("The %s secret contains the key %s", secretName, secretKeyName))
+	} else {
 		return err
 	}
 	return nil // Existing Secret found, do nothing...
 }
 
-// newSecret will create and return a new Secret instance using the given Name/Namespace.
-func newSecret(key types.NamespacedName) *corev1.Secret {
-	return &corev1.Secret{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      key.Name,
-			Namespace: key.Namespace,
-		},
-	}
-}
-
-func (r *OperatorPipelineReconciler) getPipelineInstance(meta metav1.ObjectMeta) (*certv1alpha1.OperatorPipeline, error) {
-	pipeline := types.NamespacedName{
-		Namespace: meta.Namespace,
-		Name:      meta.Name,
-	}
-	pipelineInstance := &certv1alpha1.OperatorPipeline{}
-	err := r.Client.Get(context.TODO(), pipeline, pipelineInstance)
+func (r *OperatorPipelineReconciler) getPipeline(meta metav1.ObjectMeta) (*certv1alpha1.OperatorPipeline, error) {
+	pipelineName := newNamespacedName(meta.Name, meta.Namespace)
+	pipeline := &certv1alpha1.OperatorPipeline{}
+	err := r.Client.Get(context.TODO(), pipelineName, pipeline)
 	if err != nil {
-		if k8serrors.IsNotFound(err) {
+		if k8errors.IsNotFound(err) {
 			log.Info("pipeline resource not found. Ignoring since object must be deleted")
```

Comment:
Not sure what `Ignoring since object must be deleted` means? I've seen it in multiple places in the codebase and was wondering if this applies in the case of secrets as we don't do any CRUD operation.

Reply:
Think this would apply in the case when a CR (in our case the pipeline) no longer exists and there is nothing to reconcile..